app/destination/page.js

```diff
@@ -1,54 +1,60 @@
 "use client";
 
-import { useState } from 'react';
-
-import styles from '@/components/destination/destination.module.css';
-import { AddWishlistItem } from '@/components/destination/AddWishlistItem';
-
-// TASK - React 1 week 2
-// Move this to its own file
-const PlanetWishlistItem = ({
-  name,
-  onRemove,
-  thumbnail,
-}) => {
-  return (
-    <div className={styles.wishlistItem}>
-      <img className={styles.wishlistItemThumbnail} src={thumbnail} alt="" />
-      <b>{name.toUpperCase()}</b>
-      <button onClick={onRemove}>remove</button>
-    </div>
-  );
-}
-
+import { useState } from "react";
+import styles from "@/components/destination/destination.module.css";
+import { AddWishlistItem } from "@/components/destination/AddWishlistItem";
+import PlanetWishlistItem from "./PlanetWishlistItem";
+import { PlanetCard } from "./PlanetCard";
 
 export const Destinations = () => {
+  //state
   const [selectedPlanets, onAddPlanet] = useState([]);
-
-  let isPlanetSelected = false;
-  let numberOfPlanets = 0;
+  const [numberOfPlanets, setNumberOfPlanets] = useState(0);
+  const [isPlanetSelected, setIsPlanetSelected] = useState(false);
 
   const onAddOrRemovePlanet = (name, index) => {
-    // TASK - React 1 week 2
-    // Implement this function
-    // If you press the "ADD PLANET" the selected planet should display "SELECTED"
-    // And the counter should update, how many planets are selected (numberOfPlanets)
-    console.log(`You seleceted the following planet: ${name}, with the index of ${index}`);
-  }
+    if (selectedPlanets.includes(name)) {
+      const updatedPlanets = selectedPlanets.filter(
+        (planet) => planet !== name
+      );
+      onAddPlanet(updatedPlanets);
+
+      setNumberOfPlanets(updatedPlanets.length);
+      setIsPlanetSelected((prevState) => ({
+        ...prevState,
+        [name]: false,
+      }));
+    } else {
+      // If the planet is not selected, add it to the array
+      const updatedPlanets = [...selectedPlanets, name];
+      onAddPlanet(updatedPlanets);
+
+      setNumberOfPlanets(updatedPlanets.length);
+      setIsPlanetSelected((prevState) => ({
+        ...prevState,
+        [name]: true,
+      }));
+    }
+
+    console.log(
+      `You selected the following planet: ${name}, with the index of ${index}`
+    );
+  };
 
   return (
     <div className="fullBGpicture">
       <main className="mainContent">
         <h1>Travel destinations</h1>
         <section className="card">
           <h2>Wishlist</h2>
-          {/* TASK - React 1 week 2 */}
-          {/* Display the number Of selected planets */}
-          {/* Display the "no planets" message if it is empty! */}
-          <p>No planets in wishlist :(</p>
-          <p>You have {numberOfPlanets} in your wishlist</p>
+
+          {numberOfPlanets > 0 ? (
+            <p>You have {numberOfPlanets} in your wishlist</p>
+          ) : (
+            <p>No planets in wishlist :(</p>
```

Comment:
This is a very common practice all over the internet, just imagine opening your cart online and it will either:
- Display the items
- Have an icon and say your cart is empty